modulemd/modulemd-eol.c

```diff
@@ -0,0 +1,1196 @@
+      break;
+    case PROP_MESSAGE:
+      modulemd_eol_set_message (self, g_value_get_string (value));
+      break;
+    case PROP_OBSOLETED_BY_MODULE_NAME:
+      modulemd_eol_set_obsoleted_by_module_name (self,
+                                                 g_value_get_string (value));
+      break;
+    case PROP_OBSOLETED_BY_MODULE_STREAM:
+      modulemd_eol_set_obsoleted_by_module_stream (self,
+                                                   g_value_get_string (value));
+      break;
+    default: G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+    }
+}
+
+static void
+modulemd_eol_class_init (ModulemdEolClass *klass)
+{
+  GObjectClass *object_class = G_OBJECT_CLASS (klass);
+
+  object_class->finalize = modulemd_eol_finalize;
+  object_class->get_property = modulemd_eol_get_property;
+  object_class->set_property = modulemd_eol_set_property;
+
+  properties[PROP_VERSION] = g_param_spec_uint64 (
+    "version",
+    "Version",
+    "The metadata version of this eol object.",
+    0,
+    G_MAXUINT64,
+    0,
+    G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS | G_PARAM_CONSTRUCT_ONLY);
+
+  properties[PROP_MODIFIED] = g_param_spec_uint64 (
+    "modified",
+    "Modified",
+    "The last modified time represented as a 64-bit integer "
+    "(such as 201807011200)",
+    0,
+    G_MAXUINT64,
+    0,
+    G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS);
+
+  properties[PROP_MODULE_NAME] = g_param_spec_string (
+    "module-name",
+    "Module name",
+    "The name of the module to which these eol apply.",
+    O_DEFAULT_STRING,
+    G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS | G_PARAM_CONSTRUCT_ONLY);
+
+  properties[PROP_RESET] = g_param_spec_boolean (
+    "override-previous",
+    "Override previous",
+    "A boolean option to cancel/reset all previously specified eols.",
+    FALSE,
+    G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS | G_PARAM_CONSTRUCT);
+
+  properties[PROP_MODULE_STREAM] = g_param_spec_string (
+    "module-stream",
+    "Module stream",
+    "The name of the module stream to which these eol apply.",
+    O_DEFAULT_STRING,
+    G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS | G_PARAM_CONSTRUCT_ONLY);
+
+  properties[PROP_MODULE_CONTEXT] = g_param_spec_string (
+    "module-context",
+    "Module context",
+    "The name of the module context to which these eol apply.",
+    NULL,
+    G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS | G_PARAM_CONSTRUCT);
+
+  properties[PROP_EOL_DATE] = g_param_spec_uint64 (
+    "eol-date",
+    "EOL date",
+    "A string representing UTC date in ISO 8601 format: YYYY-MM-DD[T ]HH:MMZ",
+    0,
+    G_MAXUINT64,
+    0,
+    G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS | G_PARAM_CONSTRUCT);
+
+  properties[PROP_MESSAGE] = g_param_spec_string (
+    "message",
+    "Message",
+    "A string describing the change, reason, etc.",
+    O_DEFAULT_STRING,
+    G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS | G_PARAM_CONSTRUCT_ONLY);
```

Comment:
Why is this `G_PARAM_CONSTRUCT_ONLY`?

Reply:
I thought since the message is mandatory it can only be set upon object construction just like module name, module stream etc

Reply:
You're right, I missed that.